node/src/options.rs

```diff
@@ -79,6 +79,14 @@ pub struct RunOptions {
     #[clap(long)]
     pub p2p_outbound_connection_timeout: Option<u64>,
 
+    /// Enable multicast DNS peer discovery
+    #[clap(long)]
+    pub enable_mdns: Option<bool>,
```

Comment:
I would rather use `p2p_enable_mdns` here. It isn't ideal, but as the command line options aren't hierarchical it makes sense to use prefixes for subsystems-related options.

Reply:
I was debating between whether the p2p prefix should or should not be here. It's really inconvenient if it has to be added for each flag even if there will be no conflict, just like ike in this case. For example, `max_orphan_blocks` is unambiguous just like `mdns_query_interval` whereas `addr` isnt.

Reply:
Personally I prefer consistent naming, but I'm not going to insist. 🙃

Reply:
Sorry for bringing this up again, but as far as I can see, the options are sorted alphabetically. So without prefixes the order will be somewhat confusing.

Reply:
I'm not sure what's the best way to fix them. I change the order of these new options to be in the correct place, that is at least what Substrate seems to be doing but I take it you'd rather have the p2p prefix?

Reply:
Perhaps I'm missing something, but it looks like the order of the filed in a struct isn't important and clap outputs them in the alphabetical order. So yes, I'm talking about the prefix again. 🙃 Maybe I'm exaggerating this issue, but for me it seems better to have some structure here, so a user can easily see which options are related to some subsystem. Still it is just my opinion and I'm not going to insist (too much).

Reply:
I see your point and I like consistency too but for me it feels really awkward to type `--p2p-mdns-query-interval`.

Maybe @TheQuantumPhysicist or @muursh can make the decision since we're in a stalemate :smiley:

Reply:
We either use a prefix for everything or for nothing. Half and half is confusing.

Reply:
I think that we are forced to used prefix for some parameters that have generic name such as "address" .

Reply:
But `max_db_commit_attempts` and `max_orphan_blocks` don't have a `chainstate_` prefix. IMO P2P/RPC addresses are a special case which shouldn't dictate the entire naming scheme for command line parameters

Reply:
> But `max_db_commit_attempts` and `max_orphan_blocks` don't have a chainstate_ prefix.

It is my oversight. 🙃

Reply:
Nobody asked for my opinion but I'm going to throw it in anyway :laughing:. Whatever prefixes end up being there, I personally don't think they need or should be based on subsystem. How is the node broken down into subsystems internally I prefer to consider an implementation detail. Although we already use subsystem-based prefixes in RPC and config file too I believe.